goose.go

```diff
@@ -321,10 +335,12 @@ func (ctx Ctx) coqType(e ast.Expr) coq.Type {
 		return ctx.ptrType(e)
 	case *ast.InterfaceType:
 		if isEmptyInterface(e) {
-			return coq.TypeIdent("anyT")
+			return coq.TypeIdent("interfaceT")
 		} else {
-			ctx.unsupported(e, "non-empty interface")
+			return ctx.interfaceType(e)
 		}
+	case *ast.FuncType:
+		return coq.TypeIdent("funcT")
```

Comment:
I don't see such a definition in Perennial, so how does this compile in Coq? For now it's fine to add `funcT` to the type system without input/return types, but if we ever wanted to typecheck GooseLang programs we would need them. If you can I would translate those types even if the Coq definition ignores them.

Reply:
This is the only thing preventing me from merging, could you look into adding this definition to Perennial?

Reply:
Resolving due to updated PR